projects/com.oracle.truffle.llvm.parser.factories/src/com/oracle/truffle/llvm/parser/factories/LLVMAggregateFactory.java

```diff
@@ -101,7 +102,38 @@ public static LLVMExpressionNode createExtractValue(LLVMBaseType type, LLVMAddre
         }
     }
 
-    public static Node createStructWriteNode(LLVMExpressionNode parsedConstant, ResolvedType resolvedType) {
+    public static LLVMExpressionNode createInsertValue(LLVMAddressNode resultAggregate, LLVMAddressNode sourceAggregate, int size, int offset, LLVMExpressionNode valueToInsert,
+                    LLVMBaseType llvmType) {
+        switch (llvmType) {
+            case FLOAT:
+                return new LLVMInsertFloatValueNode(resultAggregate, sourceAggregate, size, offset, (LLVMFloatNode) valueToInsert);
+            case DOUBLE:
+                return new LLVMInsertDoubleValueNode(resultAggregate, sourceAggregate, size, offset, (LLVMDoubleNode) valueToInsert);
+            default:
+                throw new AssertionError(llvmType);
+        }
+    }
+
+    public static LLVMExpressionNode createStructConstantNode(LLVMParserRuntime runtime, boolean packed, int structSize, ResolvedType[] types, LLVMExpressionNode[] constants) {
+        int[] offsets = new int[types.length];
+        LLVMStructWriteNode[] nodes = new LLVMStructWriteNode[types.length];
+        int currentOffset = 0;
+        // FIXME alignment
```

Comment:
Maybe add a more expressive FIXME so someone else can work on it if they have time?